man/sd_event_source_set_ratelimit.xml

```diff
@@ -0,0 +1,150 @@
+  "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!-- SPDX-License-Identifier: LGPL-2.1-or-later -->
+
+<refentry id="sd_event_source_set_ratelimit" xmlns:xi="http://www.w3.org/2001/XInclude">
+
+  <refentryinfo>
+    <title>sd_event_source_set_ratelimit</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>sd_event_source_set_ratelimit</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_event_source_set_ratelimit</refname>
+    <refname>sd_event_source_get_ratelimit</refname>
+    <refname>sd_event_source_is_ratelimited</refname>
+
+    <refpurpose>Configure rate limiting on event sources</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-event.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_set_ratelimit</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+        <paramdef>uint64_t <parameter>interval_usec</parameter></paramdef>
+        <paramdef>unsigned <parameter>burst</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_get_ratelimit</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+        <paramdef>uint64_t* <parameter>ret_interval_usec</parameter></paramdef>
+        <paramdef>unsigned* <parameter>ret_burst</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_is_ratelimited</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+      </funcprototype>
+
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_event_source_set_ratelimit()</function> may be used to enforce rate limiting on an
+    event source. When used an event source will be temporarily turned off when it fires more often then a
+    specified burst number within a specified time interval. This is useful as simple mechanism to avoid
+    event source starvation if high priority event sources fire very frequently and monopolize
+    runtime.</para>
+
+    <para>Pass the event source to operate on as first argument, a time interval in microseconds as second
+    argument and a maximum dispatch limit ("burst") as third parameter. Whenever the event source is
+    dispatched more often than the specified burst within the specified interval it is placed in a mode
+    similar to being disabled with
+    <citerefentry><refentrytitle>sd_event_source_set_enabled</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    and the <constant>SD_EVENT_OFF</constant> parameter. However it is disabled only temporarily – once the
+    specified interval is over regular operation resumes — until the specified rate limiting is hit the next
```

Comment:
I'm not sure what this "until" refers to. It'd seem to be "it is disabled only temporarrily ... until the limiting is hit the next time", but that doesn't make sense.